ckan/prerun.py

```diff
@@ -0,0 +1,185 @@
+import os
+import sys
+import subprocess
+import psycopg2
+import urllib2
+import re
+
+import time
+
+ckan_ini = os.environ.get('CKAN_INI', '/srv/app/production.ini')
+
+RETRY = 5
+
+def check_db_connection(retry=None):
+
+    print '[prerun] Start check_db_connection...'
+
+    if retry is None:
+        retry = RETRY
+    elif retry == 0:
+        print '[prerun] Giving up after 5 tries...'
+        sys.exit(1)
+
+    conn_str = os.environ.get('CKAN_SQLALCHEMY_URL', '')
+    try:
+        connection = psycopg2.connect(conn_str)
+
+    except psycopg2.Error as e:
+        print str(e)
+        print '[prerun] Unable to connect to the database...try again in a while.'
+        import time
+        time.sleep(10)
+        check_db_connection(retry = retry - 1)
+    else:
+        connection.close()
+
+def check_solr_connection(retry=None):
+
+    print '[prerun] Start check_solr_connection...'
+
+    if retry is None:
+        retry = RETRY
+    elif retry == 0:
+        print '[prerun] Giving up after 5 tries...'
+        sys.exit(1)
+
+    url = os.environ.get('CKAN_SOLR_URL', '')
+    search_url = '{url}/select/?q=*&wt=json'.format(url=url)
+
+    try:
+        connection = urllib2.urlopen(search_url)
+    except urllib2.URLError as e:
+        print str(e)
+        print '[prerun] Unable to connect to solr...try again in a while.'
+        import time
+        time.sleep(10)
+        check_solr_connection(retry = retry - 1)
+    else:
+        import re
+        conn_info = connection.read()
+        conn_info = re.sub(r'"zkConnected":true', '"zkConnected":True', conn_info)
+        eval(conn_info)
+
+def init_db():
+
+    print '[prerun] Start init_db...'
+
+    db_command = ['paster', '--plugin=ckan', 'db', 'init', '-c', ckan_ini]
+    validation_command = ['paster', '--plugin=ckanext-validation', 'validation', 'init-db', '-c', ckan_ini]
```

Comment:
@osaetinevbuoma Biskit doesn't use the `ckanext-validation` plugin, so there is no need for `prerun.py`.

Reply:
I'm closing this PR.